data/repository/src/commonMain/kotlin/social/androiddev/common/repository/timeline/TimelineRepoModule.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * This file is part of Dodo.
+ *
+ * Dodo is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * Dodo is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Dodo. If not, see <https://www.gnu.org/licenses/>.
+ */
+package social.androiddev.common.repository.timeline
+
+import com.squareup.sqldelight.runtime.coroutines.asFlow
+import com.squareup.sqldelight.runtime.coroutines.mapToList
+import kotlinx.coroutines.flow.map
+import org.koin.core.module.Module
+import org.koin.dsl.module
+import org.mobilenativefoundation.store.store5.Bookkeeper
+import org.mobilenativefoundation.store.store5.Market
+import org.mobilenativefoundation.store.store5.NetworkFetcher
+import org.mobilenativefoundation.store.store5.NetworkUpdater
+import org.mobilenativefoundation.store.store5.OnNetworkCompletion
+import org.mobilenativefoundation.store.store5.Store
+import social.androiddev.common.network.MastodonApi
+import social.androiddev.common.network.model.Status
+import social.androiddev.common.persistence.localstorage.DodoAuthStorage
+import social.androiddev.common.timeline.TimelineDatabase
+import social.androiddev.common.timeline.TimelineItem
+
+/**
+ * Koin module containing all koin/bean definitions for
+ * timeline repository delegates.
+ */
+val timelineRepoModule: Module = module {
+
+    factory<HomeTimelineRepository> { RealHomeTimelineRepository(get()) }
+
+    factory<Store<FeedType, List<TimelineItem>, List<TimelineItem>>> {
+        val database = get<TimelineDatabase>()
+        Store.by(
```

Comment:
So Im not familiar with Store, but seems like there is a lot of logic going on in this koin module. Should a lot of the logic in this file be moved with a UseCase, which takes in a parameter(s) and then know's what feed to build/query?

Reply:
I'll move the logic to extension functions/static functions that we can test on their own. I'll inject single use cases/delegates here. For example the reader should just be something like `get<TimelineQueries>().feedFor(feedType)`

Pardon the dust and thank you for the callout. 

I'll add tests everywhere before landing this